src/autoadvance.js

```diff
@@ -0,0 +1,42 @@
+module.exports = function autoadvance(player, timeout) {
+  var resetadvance = function resetadvance() {
+    if (player.playlist._timeoutId) {
+      window.clearTimeout(player.playlist._timeoutId);
+    }
+
+    if (player.playlist._ontimeout) {
+      player.off('ended', player.playlist._ontimeout);
+    }
+
+    player.playlist._timeoutId = null;
+    player.playlist._ontimeout = null;
+  };
+
+  // we are want to cancel the auto advance or accidentally called it with a bogus value
+  if (typeof timeout !== 'number' || timeout < 0) {
```

Comment:
We should make sure that `NaN`, `Infinity`, and `-Infinity` behave as resets for autoadvance. Also, we should should make sure those values don't break other places where we type check against `"number"`.